packages/teleterm/src/ui/StatusBar/ShareFeedback/ShareFeedbackForm.tsx

```diff
@@ -1,83 +1,124 @@
+}
+
+export function ShareFeedbackForm(props: ShareFeedbackProps) {
+  const isSubmitInProgress =
+    props.submitFeedbackAttempt.status === 'processing';
+
+  return (
+    <Flex bg="primary.main" p={3} borderRadius={3} maxWidth="370px">
+      <Validation>
+        {({ validator }) => (
+          <Flex
+            flexDirection="column"
+            as="form"
+            onSubmit={e => {
+              e.preventDefault();
+              if (validator.validate()) {
+                props.submitFeedback();
+              }
+            }}
+          >
+            <Flex justifyContent="space-between" mb={2}>
+              <Text typography="h4" bold color="text.primary">
+                Provide your feedback
+              </Text>
+              <ButtonIcon
+                type="button"
+                onClick={props.onClose}
+                title="Close"
+                color="text.secondary"
+              >
+                <Close fontSize={5} />
+              </ButtonIcon>
+            </Flex>
+            <Link
+              href="https://github.com/gravitational/teleport/issues/new?assignees=&labels=bug&template=bug_report.md"
+              target="_blank"
+            >
+              Submit a Bug
+            </Link>
+            <Link href="https://goteleport.com/signup/" target="_blank">
+              Try Teleport Cloud
+            </Link>
+            {props.submitFeedbackAttempt.status === 'error' && (
+              <SubmissionError
+                submitFeedbackAttempt={props.submitFeedbackAttempt}
+              />
+            )}
+            {props.submitFeedbackAttempt.status === 'success' ? (
+              <Alerts.Success mt={3} mb={0}>
+                Your feedback has been submitted. Thank you!
+              </Alerts.Success>
+            ) : (
+              <>
+                <ShareFeedbackFormFields
+                  disabled={isSubmitInProgress}
+                  formValues={props.formValues}
+                  setFormValues={props.setFormValues}
+                />
+                <ButtonPrimary
+                  disabled={isSubmitInProgress}
+                  block
+                  type="submit"
+                  mt={4}
+                >
+                  Submit
+                </ButtonPrimary>
+              </>
+            )}
+          </Flex>
+        )}
+      </Validation>
+    </Flex>
+  );
 }
 
-export function ShareFeedbackForm({
-  formValues,
-  setFormValues,
-}: ShareFeedbackFormProps) {
-  function updateFormField<T extends keyof ShareFeedbackFormValues>(
-    field: T,
-    value: ShareFeedbackFormValues[T]
-  ) {
-    setFormValues({ ...formValues, [field]: value });
+function SubmissionError(props: { submitFeedbackAttempt: Attempt<Response> }) {
+  function getErrorText() {
+    if (props.submitFeedbackAttempt.statusText === FEEDBACK_TOO_LONG_ERROR) {
+      return (
+        <span>
+          That's a very long reason why. Please let us know more in{' '}
```

Comment:
```suggestion
          That's a very long reason suggestion. Please let us know more in{' '}
```

Reply:
Done, I think you meant `That's a very long suggestion.` :)